WooCommerce/Classes/ViewRelated/Products/Edit Product/Product Settings/ProductSettingsRows.swift

```diff
@@ -102,10 +102,12 @@ enum ProductSettingsRows {
         }
 
         func handleTap(sourceViewController: UIViewController, onCompletion: @escaping (ProductSettings) -> Void) {
-            let passwordProtectedAvailable = ServiceLocator.stores.isAuthenticatedWithoutWPCom == false
-            /// If the password was not fetched for user authenticated with WPCom,
-            /// the cell is not selectable
-            if settings.password == nil && passwordProtectedAvailable {
+            /// Check if the store is eligible for password editing or is authenticated without WPCom.
```

Comment:
Should this comment say "is authenticated with WPCom" instead?

And with that being said, wdyt about moving the `.isAuthenticatedWithoutWPCom == false` check to the first case just as a bit of optimization (remove the need to check db if auth with wpcom site).

Reply:
Good catch and good suggestion. I've made the change here: 6533335804d45842b3ced53c78e50d4d48a0f035